front/app/components/Form/Components/Controls/RatingControl.tsx

```diff
@@ -0,0 +1,178 @@
+        break;
+      case 'ArrowRight':
+      case 'ArrowUp':
+        newValue = Math.min(maximum, value + 1);
+        break;
+      case 'Home':
+        newValue = minimum;
+        break;
+      case 'End':
+        newValue = maximum;
+        break;
+      default:
+        return;
+    }
+
+    handleChange(path, newValue);
+    if (sliderRef.current) {
+      sliderRef.current.setAttribute('aria-valuenow', String(newValue));
+    }
+    event.preventDefault();
+  };
+
+  return (
+    <>
+      <FormLabel
+        htmlFor={sanitizeForClassname(id)}
+        labelValue={getLabel(uischema, schema, path)}
+        optional={!required}
+        subtextValue={getSubtextElement(uischema.options?.description)}
+        subtextSupportsHtml
+      />
+      {answerNotPublic && (
+        <Text mb="8px" mt="0px" fontSize="s">
+          <FormattedMessage {...messages.notPublic} />
+        </Text>
+      )}
+      <Box
+        data-testid="ratingControl"
+        role="slider"
+        ref={sliderRef}
+        aria-valuemin={minimum}
+        aria-valuemax={maximum}
+        aria-labelledby={sanitizeForClassname(id)}
+        tabIndex={0}
+        onKeyDown={handleKeyDown}
+      >
+        <Box
+          gap={isSmallerThanPhone ? '4px' : '8px'}
+          display="flex"
+          flexWrap="wrap"
+          justifyContent="center"
+        >
+          {[...Array(maximum).keys()].map((i) => {
+            const visualIndex = i + 1;
+            return (
+              <Box
+                flexBasis={100 / maximum}
+                key={`${path}-radio-${visualIndex}`}
+                minWidth={getButtonWidth()}
+              >
+                <Button
+                  py="12px"
+                  id={`${sanitizeForClassname(
+                    id
+                  )}-rating-option-${visualIndex}`}
+                  tabIndex={-1}
+                  aria-pressed={data === visualIndex}
+                  px="0"
+                  width="100%"
+                  onClick={() => handleChange(path, visualIndex)}
+                  buttonStyle="text"
+                >
+                  <Box
+                    display="flex"
+                    flexDirection="column"
+                    alignItems="center"
+                    gap="4px"
+                  >
+                    <Icon
+                      name={data >= visualIndex ? 'ratingFilled' : 'rating'}
+                      height="40px"
+                      width="40px"
+                      fill={
+                        data >= visualIndex
+                          ? theme.colors.tenantPrimary
+                          : theme.colors.tenantPrimaryLighten75
+                      }
+                    />
+                    {visualIndex}
+                  </Box>
+                </Button>
+              </Box>
+            );
+          })}
+        </Box>
+        <VerificationIcon show={uischema.options?.verificationLocked} />
```

Comment:
Is this field able to be used as a User custom field? If not then it probably does not need the verification lock